src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -318,24 +318,12 @@ void primitive_inst::update_shape() {
         }
     }
 
+    // Reset shape is defined by input shape changes
     if (input_shape_changed)
         set_shape_change();
     else
         reset_shape_change();
 
-    // We assume that tensor ranks are static, thus shape_of doesn't need to update anything even if input shape is dynamic
-    if (_node->is_type<shape_of>() && !input_shape_changed) {
-        reset_shape_change();
-        return;
```

Comment:
Please keep the early return call for both shape_of and loop operations, as there is no need to execute the code below for them if the input shape hasn’t changed